pkg/api/validation/validation.go

```diff
@@ -373,9 +398,11 @@ func ValidateObjectMetaUpdate(newMeta, oldMeta *api.ObjectMeta, fldPath *field.P
 	allErrs = append(allErrs, ValidateImmutableField(newMeta.Namespace, oldMeta.Namespace, fldPath.Child("namespace"))...)
 	allErrs = append(allErrs, ValidateImmutableField(newMeta.UID, oldMeta.UID, fldPath.Child("uid"))...)
 	allErrs = append(allErrs, ValidateImmutableField(newMeta.CreationTimestamp, oldMeta.CreationTimestamp, fldPath.Child("creationTimestamp"))...)
+	allErrs = append(allErrs, ValidateImmutableField(newMeta.Finalizers, oldMeta.Finalizers, fldPath.Child("finalizers"))...)
```

Comment:
> I think that update is strictly more powerful than delete, and therefore it's not surprising at all that update permission allows deletes; you guys seem to think the opposite? Is there prior art here? I'm not sure how you're going to convince me that power to update should not include power to delete.

I see the different permissions as different without a clear covers relationship amongst all of them.  Bearing in mind that our current API objects only represent a few enumerated cases, not the full set of all possible ones I see this:

1. List - covers Watch (repeated List) and Get.
2. Create - does not cover anything.  The power to create implies the power to set any "set-once" fields that are immutable after creation.  It also results in quota charges.
3. Patch - does not cover anything.  The power to Patch does not imply the power to Create (patch can't change immutable fields and can't charge quota).  The power to Patch does not imply the power to Get (patch can blindly set fields, `kubectl annotate` as a for instance.)
4. Delete - does not cover anything.  The power to Delete doesn't imply that you Get (or List or Watch) the thing you're deleting (maybe its a timed-based secret deletion taking a batch list and shouldn't ever be allowed to see a secret).  It doesn't imply Create (can't set immutable fields) or Patch or Update (can't change the meaning of the resource.)
5. Update - covers Get (not List or Watch) and Patch.  Update is per-item and requires a resourceVersion, so to make use of Update, you had the original object at some point in time.  Update does not cover Create because it can't set immutable fields or charge quota.  Update does not cover Delete since it doesn't replenish quota and (given immutable fields) doesn't necessarily imply the power to destroy data.